tests/cases/compiler/-da/divergentAccessorsTypes-56894-b.ts

```diff
@@ -0,0 +1,44 @@
+
+type T12 = {
+    a: 1|2;
+    set gsa(x: 1|2);
+    get gsa(): 1|2;
+    set sb(x: 1|2);
+    get gc(): 1|2;
+};
+type T23 = {
+    a: 2|3;
+    set gsa(x: 2|3);
+    get gsa(): 2|3;
+    set sb(x: 2|3);
+    get gc(): 2|3;
+};
+
+function fu(u:T12|T23) {
+    u.a = 1;
+    u.sb = 1;
+    u.gsa = 1;
+    u.a = 2;
+    u.sb = 2;
+    u.gsa = 2;
+    u.a = 3;
+    u.sb = 3;
+    u.gsa = 3;
+    if (u.gsa === 1) {
+        u.gsa;
+    }
+    if (u.gc === 1) {

```

Comment:
u.gsa===1 produces a read range error but u.gc === 1 does not.  That's not an intuitively obvious rule.